### ISSUE TITLE
Allow DNS Resolution to be user configurable

### ISSUE THREAD

**Issue by Gelob** [ntop/ntopng#2297]
Would it be possible to consider having DNS resolution be user configurable? We have cases where some users are using ntopng for security/forensic analysis purposes and they don't want to see the DNS name for an IP address when viewing historical flows because that may not have been the DNS name at the time the flow occurred. Other users that are network operators viewing real time data for top talkers, etc find having the DNS name useful.

**Comment by simonemainardi** [comment]
please consider using

```
[--dns-mode|-n] <mode>              | DNS address resolution mode
                                    | 0 - Decode DNS responses and resolve
                                    |     local numeric IPs only (default)
                                    | 1 - Decode DNS responses and resolve all
                                    |     numeric IPs
                                    | 2 - Decode DNS responses and don't
                                    |     resolve numeric IPs
                                    | 3 - Don't decode DNS responses and don't
                                    |     resolve numeric IPs
```

You can disable DNS resolution if not necessary.

**Comment by simonemainardi** [comment]
seems `--dns-mode` is ignored when visualizing historical data, causing addresses to always be resolved. will address this.

**Comment by Gelob** [comment]
@simonemainardi I look forward to that being fixed. I kn ow of the --dns-mode option but that is global. I know making this user configurable may not be possible or accepted now but I just wanted to put the request out there for the future should it be re-considered or the DNS code changed.

**Comment by simonemainardi** [comment]
dns setting must be honored when browsing historical data. Specifically, if dns resolution is off, no dns query has to be performed.

**Comment by emanuele-f** [comment]
Please note that this was fixed then reverted after internal discussion:

Fix: ae04ad2c7c4d2bc2792530a52c394f9c38cd1716
Revert: e597592de2f218e4710a4c04ac0841bfecf17e92

**Comment by simonemainardi** [comment]
this is slightly different. What I am asking is to disable DNS resolution when `--dns-mode` is 2 or 3
